src/proper.erl

```diff
@@ -1729,24 +1741,34 @@ apply_skip(Args, Prop) ->
 -spec report_imm_result(imm_result(), opts()) -> 'ok'.
 report_imm_result(#pass{samples = Samples, printers = Printers,
 			performed = Performed},
-		  #opts{expect_fail = ExpectF, output_fun = Print}) ->
-    case ExpectF of
-	true ->
-	    Print("\033[1;31mFailed: All tests passed when a failure was expected.\033[0m~n", []);
-	false ->
-	    Print("\033[1;32mOK: Passed ~b test(s).~n\033[0m", [Performed])
+                  #opts{expect_fail = ExpectF, output_fun = Print,
+                        nocolors = NoColors}) ->
+    case {ExpectF, NoColors} of
+        {true, true} ->
```

Comment:
I suggest that you use two nested `case`s here rather than wrapping `ExpectF` and `NoColors` to a tuple and having a `case` with four branches.